src/Mailgun/Api/MailingList.php

```diff
@@ -0,0 +1,307 @@
+     *
+     * @throws \Exception
+     */
+    public function delete($address)
+    {
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpDelete(sprintf('/v3/lists/%s', $address));
+
+        return $this->hydrateResponse($response, DeleteResponse::class);
+    }
+
+    /**
+     * Returns a paginated list of members of the mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param int    $limit      Maximum number of records to return (optional: 100 by default)
+     * @param mixed  $subscribed `yes` to lists subscribed, `no` for unsubscribed. list all if null
+     *
+     * @return MembersResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function members($address, $limit = 100, $subscribed = null)
+    {
+        Assert::stringNotEmpty($address);
+        Assert::natural($limit);
+        Assert::oneOf($subscribed, [null, 'yes', 'no']);
+
+        $params = [
+            'limit' => $limit,
+            'subscribed' => $subscribed,
+        ];
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s/members/pages', $address), $params);
+
+        return $this->hydrateResponse($response, MembersResponse::class);
+    }
+
+    /**
+     * Shows a single member of the mailing list.
+     *
+     * @param string $list    Address of the mailing list
+     * @param string $address Address of the member
+     *
+     * @return ShowMemberResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function showMember($list, $address)
+    {
+        Assert::stringNotEmpty($list);
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s/members/%s', $list, $address));
+
+        return $this->hydrateResponse($response, ShowMemberResponse::class);
+    }
+
+    /**
+     * Creates (or updates) a member of the mailing list.
+     *
+     * @param string $list       Address of the mailing list
+     * @param string $address    Address for the member
+     * @param string $name       Name for the member (optional)
+     * @param array  $vars       Array of field => value pairs to store additional data
+     * @param string $subscribed `yes` to add as subscribed (default), `no` as unsubscribed
+     * @param string $upsert     `yes` to update member if present, `no` to raise error in case of a duplicate member (default)
+     *
+     * @return CreateMemberResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function createMember($list, $address, $name = null, $vars = [], $subscribed = 'yes', $upsert = 'no')
+    {
+        Assert::stringNotEmpty($list);
+        Assert::stringNotEmpty($address);
+        Assert::nullOrStringNotEmpty($name);
+        Assert::isArray($vars);
+        Assert::oneOf($subscribed, ['yes', 'no']);
+        Assert::oneOf($upsert, ['yes', 'no']);
+
+        $vars = json_encode($vars);
```

Comment:
Although it works, personally I don't like to reuse the same variable to change multiple types. At the beginning, this variable is an `array`, and then it's converted to a JSON `string` (assuming that the `json_encode` function has worked).

I would prefer to use a 2nd variable to store the encoded content (like `$jsonVars` to keep it simple), and then [add a checker](https://secure.php.net/manual/en/function.json-last-error.php) to confirm the last JSON method didn't throw an error. If an error was returned then we should throw our Exception to alert the developer.

Reply:
On a 2nd thought I am not sure if this is useless, because the Payload sent to the API should be converting all parameters to a single JSON string, including the arguments. I think this line doesn't make sense at the current stage.